onyo/lib/onyo.py

```diff
@@ -510,6 +512,32 @@ def _fsck_clean_tree(self) -> bool:
 
         return True
 
+    def _fsck_no_pseudo_keys(self) -> bool:
+        """
+        Check that no asset contains any pseudo-key names.
+        """
+        assets_with_pseudo_keys = {}
+        pseudo_keys = ["type", "make", "model", "serial"]
+
+        for asset in self.assets:
```

Comment:
The rest of the for-loop can be simplified to:
```
violation_list = [x for x in pseudo_keys if x in self._read_asset(asset)]
if violation_list:
    assets_with_pseudo_keys[asset] = violation_list
```

Reply:
Not that this looks to be a costly part of the code, but the list comprehension also reduces the number of assignments and appends dramatically.